### PR TITLE
DropDownEditor FIX: DataGrid reverts changes in batch mode if the dropdown icon is clicked(T823431)

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

**Review by churkin — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js [hunk 1]
```diff
@@ -512,19 +512,32 @@ QUnit.test("native focus event should not be triggered if dropdown button clicke
...
+QUnit.test("focusout should be fired after click on the dropDownButton when editor isn't focused (T823431)", function(assert) {
+    var $dropDownEditor = $("#dropDownEditorLazy").dxDropDownEditor({
+            focusStateEnabled: true
+        }),
+        $dropDownButton = $dropDownEditor.find(".dx-dropdowneditor-button");
+
+    var e = $.Event("mousedown");
+    $dropDownButton.trigger(e);
+
+    assert.notOk(e.isDefaultPrevented(), "focusout was prevented");
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]: "focusout was NOT prevented"

File: testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js [hunk 2]
```diff
@@ -512,19 +512,32 @@ QUnit.test("native focus event should not be triggered if dropdown button clicke
     assert.equal(focusinHandler.callCount, 0, "native focus should not be triggered");
 });
 
-QUnit.test("focusout should not be fired after click on the dropDownButton", function(assert) {
+QUnit.test("focusout should not be fired after click on the dropDownButton when editor is focused", function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]:
```suggestion
QUnit.testInActiveWindow("focusout should not be fired after click on the dropDownButton when editor is focused", function(assert) {
```

File: testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js [hunk 3]
```diff
@@ -512,19 +512,32 @@ QUnit.test("native focus event should not be triggered if dropdown button clicke
...
-    var e = $.Event("mousedown");
+    $dropDownEditor.dxDropDownEditor("focus");
 
+    var e = $.Event("mousedown");
     $dropDownButton.trigger(e);
 
     assert.ok(e.isDefaultPrevented(), "focusout was prevented");
 });
 
+QUnit.test("focusout should be fired after click on the dropDownButton when editor isn't focused (T823431)", function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]:
```suggestion
QUnit.testInActiveWindow("focusout should be fired after click on the dropDownButton when editor isn't focused (T823431)", function(assert) {
```

File: js/ui/drop_down_editor/ui.drop_down_button.js [hunk 4]
```diff
@@ -17,7 +17,11 @@ export default class ClearButton extends TextEditorButton {
             !editor.option("openOnFieldClick") && editor._openHandler(e);
         });
 
-        eventsEngine.on(instance.$element(), "mousedown", (e) => { e.preventDefault(); });
+        eventsEngine.on(instance.$element(), "mousedown", (e) => {
+            if($(editor.element()).is(".dx-state-focused")) {
```
Comment (San4es) [inline js/ui/drop_down_editor/ui.drop_down_button.js]:
```suggestion
            if(editor.$element().is(".dx-state-focused")) {
```

**Review by San4es — DISMISSED** [review]
LGTM

File: testing/functional/model/internal/dropDownList.ts [hunk 5]
```diff
@@ -26,6 +30,10 @@ export default abstract class DropDownList extends TextBox {
         return ATTR.popupId;
     }
 
+    getDropDownButton() {
```
Comment (churkin) [inline testing/functional/model/internal/dropDownList.ts]:
```javascript
dropDownButton: Selector;
...
constructor(...) {
...
    this.dropDownButton = this.element.find(`.${CLASS.dropDownButton}`);
}
```

**Review by churkin — CHANGES_REQUESTED** [review]

**Review by San4es — APPROVED** [review]

**Review by churkin — APPROVED** [review]

**Pull request merged**